lib/os/fdtable.c

```diff
@@ -389,7 +389,14 @@ int zvfs_close(int fd)
 	(void)k_mutex_lock(&fdtable[fd].lock, K_FOREVER);
 	if (fdtable[fd].vtable->close != NULL) {
```

Comment:
That check probably doesn't make much sense if `close2()` is chosen?

Reply:
It's a union, so it should be fine